flower_shop/security/record_rules.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+
+        <record id="product_gardener_rule" model="ir.rule">
+            <field name="name">Product Gardener</field>
+            <field name="model_id" ref="product.model_product_template"/>
+            <field name="domain_force">['|', ('gardeners_ids', 'in', user.id), ('gardeners_ids', '=', False)]</field>
+            <field name="perm_read">1</field>
+            <field name="perm_create">0</field>
+            <field name="perm_unlink">0</field>
+            <field name="perm_write">0</field>
+            <field name="groups" eval="[(4, ref('coreline_btco.gardner_user'))]"/>

```

Comment:
Since you have applied a group to the record rule, it means it is **Group-specific** and not **Global**, be careful when you do this as **group-specific** rules are combines with **OR** logical operators, whereas **global** rules are combined using **AND** logical operators.